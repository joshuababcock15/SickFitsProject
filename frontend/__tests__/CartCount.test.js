import { render, screen } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount />', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });
  it('Matches Snapshot', () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });
  it('updates via props', async () => {
    const { container, rerender } = render(<CartCount count={11} />);
    expect(container.textContent).toBe('11');
    // same as
    // expect(container).toHaveTextContent('11');
    rerender(<CartCount count="12" />);
    // can just use toHaveTextContent which is the react way
    // expect(container).toHaveTextContent('12');

    // using js to update the props
    expect(container.textContent).toBe('1211');
    // wait for 4 ms
    await wait(400);
    // another option below
    // await screen.findByText('12');
    expect(container.textContent).toBe('12');
    expect(container).toMatchSnapshot();
  });
});
