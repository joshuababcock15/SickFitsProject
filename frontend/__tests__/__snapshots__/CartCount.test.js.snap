// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> Matches Snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-xxvp4g-1 cNcSUx"
  >
    <div>
      <div
        class="CartCount__Dot-xxvp4g-0 lmpTsU count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount /> updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-xxvp4g-1 cNcSUx"
  >
    <div>
      <div
        class="CartCount__Dot-xxvp4g-0 lmpTsU count count-enter-done"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
